ivy.el

```diff
@@ -3990,7 +3990,14 @@ Skip buffers that match `ivy-ignore-buffers'."
 
 (defun ivy-switch-buffer-occur ()
   "Occur function for `ivy-switch-buffer' using `ibuffer'."
-  (ibuffer nil (buffer-name) (list (cons 'name ivy--old-re))))
+  (ibuffer
+   nil (buffer-name)
+   `((or ,@(cl-loop for match in (ivy--switch-buffer-matcher
+                                  ivy--old-re ivy--old-cands)
+                    if (not (eq 'ivy-virtual
+                                (get-text-property 0 'face match)))
+                    collect `(name . ,(format "\\_<%s\\_>"
+                                              (regexp-quote match))))))))
```

Comment:
Is it really necessary to use `cl-loop` when a simpler reduction will do?
```el
`((or ,@(cl-mapcan
            (lambda (match)
              (unless (eq 'ivy-virtual (get-text-property 0 'face match))
                `((name . ,(format "\\_<%s\\_>" (regexp-quote match))))))
            (ivy--switch-buffer-matcher ivy--old-re ivy--old-cands))))
```

Reply:
Sorry, this code doesn't belong in this PR. Since `ivy-make-magic-action` is broken without this fix, I thought it would get merged quickly. So I opened a PR from my master branch. That was a mistake.